### PR TITLE
Chapter 7: Sign Up

### PR DESCRIPTION
**Pull request by megatimranikram**

### CODE REVIEW

**Comment by megatimranikram** [comment]
ready

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      flash[:success] = "Welcome to the sample app!"
+      redirect_to @user
+    else
+      flash[:danger] = "Account registration failed"
+      render :new
+    end
+  end
+
+  private
+  def user_params
```
Comment (saobv-2223) [inline app/controllers/users_controller.rb]:
```suggestion
  private
  
  def user_params
```

File: config/application.rb [hunk 2]
```diff
@@ -10,4 +10,4 @@ class Application < Rails::Application
     config.i18n.available_locales = [:en, :ru]
     config.i18n.default_locale = :en
   end
-end
+end
```
Comment (saobv-2223) [inline config/application.rb]:
```suggestion
end

```

File: config/database.yml.example [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+test:
+  <<: *default
+  database: sample_app_test
+
+production:
+  <<: *default
+  database: sample_app_production
+  username: sample_app
+  password: <%= ENV['SAMPLE_APP_DATABASE_PASSWORD'] %>
```
Comment (saobv-2223) [inline config/database.yml.example]:
```suggestion
  password: <%= ENV['SAMPLE_APP_DATABASE_PASSWORD'] %>
  
```

File: config/locales/en.yml [hunk 4]
```diff
@@ -1,2 +1,2 @@
 en:
-  hello: "Hello world"
+  hello: "Hello world"
```
Comment (saobv-2223) [inline config/locales/en.yml]:
```suggestion
  hello: "Hello world"
  
```

File: db/migrate/20200825082802_add_index_to_users_email.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+class AddIndexToUsersEmail < ActiveRecord::Migration[6.0]
+  def change
+    add_index :users, :email, unique: true
+  end
+end
```
Comment (saobv-2223) [inline db/migrate/20200825082802_add_index_to_users_email.rb]:
```suggestion
end

```

File: db/schema.rb [hunk 6]
```diff
@@ -10,14 +10,15 @@
...
+  create_table "users", options: "ENGINE=InnoDB DEFAULT CHARSET=latin1", force: :cascade do |t|
     t.string "name"
     t.string "email"
     t.datetime "created_at", precision: 6, null: false
     t.datetime "updated_at", precision: 6, null: false
     t.string "password_digest"
+    t.index ["email"], name: "index_users_on_email", unique: true
   end
 
 end
```
Comment (saobv-2223) [inline db/schema.rb:23]:
```suggestion
end

```

**Comment by megatimranikram** [comment]
ready

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (megatimranikram) [reply app/controllers/users_controller.rb]: done sir

File: config/application.rb (hunk 2 above)
Comment (megatimranikram) [reply config/application.rb]: done sir

File: config/database.yml.example (hunk 3 above)
Comment (megatimranikram) [reply config/database.yml.example]: done sir

File: config/locales/en.yml (hunk 4 above)
Comment (megatimranikram) [reply config/locales/en.yml]: done sir

File: db/migrate/20200825082802_add_index_to_users_email.rb (hunk 5 above)
Comment (megatimranikram) [reply db/migrate/20200825082802_add_index_to_users_email.rb]: done sir

File: db/schema.rb (hunk 6 above)
Comment (megatimranikram) [reply db/schema.rb:23]: done sir

File: app/controllers/users_controller.rb [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    @user = User.new
+  end
+
+  def create
+    @user = User.new user_params
+    if @user.save
+      flash[:success] = "Welcome to the sample app!"
+      redirect_to @user
+    else
+      flash[:danger] = "Account registration failed"
```
Comment (vanvtt-0952) [inline app/controllers/users_controller.rb]: ` flash[:danger] = "Account registration failed"` change to ` flash.now[:danger] = "Account registration failed"`

**Comment by vanvtt-0952** [comment]
merged
plz, fixed 1 comment, then you can merge this pull request.

File: app/controllers/users_controller.rb (hunk 7 above)
Comment (megatimranikram) [reply app/controllers/users_controller.rb]: done ms

**Pull request merged**